contracts/factory/CustomSmartWalletFactory.sol

```diff
@@ -118,20 +118,17 @@ contract CustomSmartWalletFactory is ICustomSmartWalletFactory {
         bytes calldata initParams,
         bytes calldata sig
     ) external override {
-        bytes memory packed =
-            abi.encodePacked(
-                "\x19\x10",
-                owner,
-                recoverer,
-                logic,
-                index,
-                initParams
-            );
+        bytes32 _hash = keccak256(abi.encodePacked(
+            address(this),
+            owner,
+            recoverer,
+            logic,
+            index,
+            initParams
+        ));
         (sig);
-        require(
-            RSKAddrValidator.safeEquals(keccak256(packed).recover(sig), owner),
-            "Invalid signature"
-        );
+        bytes32 message = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", _hash));
```

Comment:
This change will affect the smartWalletFactoryTest(Line48) and CustomSmartWalletFactoryTest(Line50).

Reply:
@franciscotobar  Yes, indeed, thank you for pointing it out.
I've checked the failing errors and applied the required changes.